From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 9 May 2024 17:56:06 +0000
Subject: temp fixup! Content settings infrastructure

Don't even know if this will work.

Don't squash without looking into this.

Change-Id: Id2be1e0e6fb7bd9c1f694c06b63e3cf00942875c
---
 .../content_settings/core/common/content_settings.cc      | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/components/content_settings/core/common/content_settings.cc b/components/content_settings/core/common/content_settings.cc
--- a/components/content_settings/core/common/content_settings.cc
+++ b/components/content_settings/core/common/content_settings.cc
@@ -142,7 +142,13 @@ RendererContentSettingRules& RendererContentSettingRules::operator=(
     RendererContentSettingRules&& rules) = default;
 
 bool RendererContentSettingRules::operator==(
-    const RendererContentSettingRules& other) const = default;
+    const RendererContentSettingRules& other) const {
+  return std::tie(image_rules, script_rules, popup_redirect_rules,
+                  mixed_content_rules, auto_dark_content_rules) ==
+         std::tie(other.image_rules, other.script_rules,
+                  other.popup_redirect_rules, other.mixed_content_rules,
+                  other.auto_dark_content_rules);
+}
 
 ContentSettingRuleSource::ContentSettingRuleSource() = default;
 
-- 

